shared/actions/unlock-folders.js

```diff
@@ -1,39 +1,50 @@
 /* @flow */
 
+import engine from '../engine'
 import * as Constants from '../constants/unlock-folders'
-import {routeAppend} from '../actions/router'
 
-import type {RouteAppend} from '../constants/router'
 import type {TypedAsyncAction} from '../constants/types/flux'
 import type {LoadDevices, ToPaperKeyInput, CheckPaperKey, Finish} from '../constants/unlock-folders'
+import type {device_deviceList_rpc} from '../constants/types/flow-types'
 
-// TODO this won't work in a remote component until we have a synchronous way to get the latest state in a remote component.
-
-function nextPhase (): TypedAsyncAction<RouteAppend> {
-  return (dispatch, getState) => {
-    // TODO careful here since this will not be sync on a remote component!
-    const phase: string = getState().unlockFolders.phase
-    dispatch(routeAppend(phase))
-  }
-}
-
-export function loadDevice (): TypedAsyncAction<LoadDevices | RouteAppend> {
+export function loadDevice (): TypedAsyncAction<LoadDevices> {
   return dispatch => {
-    // TODO: make engine call to get devices
-    dispatch({type: Constants.loadDevices, payload: {devices: []}})
-    // Then we go to the navigate to the next phase that the reducer determined
-    dispatch(nextPhase())
+    const params : device_deviceList_rpc = {
+      method: 'device.deviceList',
+      param: {},
+      incomingCallMap: {},
+      callback: (error, devices) => {
+        if (error) {
+          console.log('Error fetching devices. Not handling this error')
+          dispatch({
+            type: Constants.loadDevices,
+            error: true,
+            payload: {error}
+          })
+        } else {
+          dispatch({
+            type: Constants.loadDevices,
+            payload: {devices}
+          })
+        }
+      }
+    }
+
+    engine.rpc(params)
```

Comment:
Ideally this calls `actions/devices` and the component just uses that state to load devices, but I think we'll probably clean up devices so I wanted to avoid coupling these things.